web/src/test/java/org/springframework/security/web/csrf/CookieCsrfTokenRepositoryTests.java

```diff
@@ -115,6 +115,30 @@ public void saveTokenNull() {
 	}
 
 	@Test
+	public void saveTokenHttpOnlyTrue() {
+		this.repository.setCookieHttpOnly(true);
+		CsrfToken token = this.repository.generateToken(this.request);
+		this.repository.saveToken(token, this.request, this.response);
+
+		Cookie tokenCookie = this.response
+				.getCookie(CookieCsrfTokenRepository.DEFAULT_CSRF_COOKIE_NAME);
+
+		assertThat(tokenCookie.isHttpOnly()).isTrue();
+	}
+
+	@Test
+	public void saveTokenHttpOnlyFalse() {
+		this.repository.setCookieHttpOnly(false);
+		CsrfToken token = this.repository.generateToken(this.request);
+		this.repository.saveToken(token, this.request, this.response);
+
+		Cookie tokenCookie = this.response
+				.getCookie(CookieCsrfTokenRepository.DEFAULT_CSRF_COOKIE_NAME);
+
+		assertThat(tokenCookie.isHttpOnly()).isFalse();
+	}
+
```

Comment:
We need a test for Servlet 3.0. The test can be written by using PowerMock and mocking the result of `ReflectionUtils.findMethod`. You can find an example in [AbstractRememberMeServicesTests](https://github.com/spring-projects/spring-security/blob/4.1.0.RC2/web/src/test/java/org/springframework/security/web/authentication/rememberme/AbstractRememberMeServicesTests.java)